source/extensions/filters/http/set_filter_state/config.cc

```diff
@@ -19,7 +20,16 @@ SetFilterState::SetFilterState(const Filters::Common::SetFilterState::ConfigShar
     : config_(config) {}
 
 Http::FilterHeadersStatus SetFilterState::decodeHeaders(Http::RequestHeaderMap& headers, bool) {
-  config_->updateFilterState({&headers}, decoder_callbacks_->streamInfo());
+  // Apply listener level configuration first.
+  config_.get()->updateFilterState({&headers}, decoder_callbacks_->streamInfo());
+
+  // If configured, apply route level configuration next.
+  auto per_route_config =
+      Http::Utility::resolveMostSpecificPerFilterConfig<Filters::Common::SetFilterState::Config>(
+          decoder_callbacks_);
+  if (per_route_config) {
+    per_route_config->updateFilterState({&headers}, decoder_callbacks_->streamInfo());
+  }
```

Comment:
Is there a reason you run both, typically a route level config will override the listener level

Reply:
I was thinking it would be nice to have common listener level ones and then have additional route specific ones. You can override by applying both (route is last so it wins).

We do have this use case - but it could also be done by just copying the listener ones into every route of course.

Reply:
I would say in that case you should have explicit options most likely, as in `override, merge, etc.` Also acceptable at first of course to just say there's only one behavior, but this would be different from most filters today afaik

Reply:
I partly agree with @EItanya . I will prefer let this filter:

1. always use the most specific config or 2. iterate all configs contains both global config, vh config, route config, etc.

It's a little weird to apply both filter level config and rotue level config, but ignore the vh config, esp they all are completely same.

Reply:
I thought I did (2). Tbh I did not know there is vhost and route level?? I will need to look into that I thought it was just 2 levels